geoutils/georaster/raster.py

```diff
@@ -917,46 +907,59 @@ def crop(
         else:
             raise ValueError("cropGeom must be a Raster, Vector, or list of coordinates.")
 
-        meta = copy.copy(self.ds.meta)
-
         if mode == "match_pixel":
-            crop_bbox = Polygon([(xmin, ymin), (xmax, ymin), (xmax, ymax), (xmin, ymax)])
-
-            crop_img, tfm = rio.mask.mask(self.ds, [crop_bbox], crop=True, all_touched=True)
-            meta.update(
-                {
-                    "height": crop_img.shape[1],
-                    "width": crop_img.shape[2],
-                    "transform": tfm,
-                }
+            # crop_bbox = Polygon([(xmin, ymin), (xmax, ymin), (xmax, ymax), (xmin, ymax)])
```

Comment:
All of this seems like a lot of changes. I'm worried that some of those changes might break stuff or introduce errors. I know we have a test suite, but as we've seen in the past, it's not bullet-proof ;-)
Why can't we use rasterio's functions anymore here? If a DatasetReader is needed, can't we even just temporarily open it?

Reply:
Because then we have to write a whole memoryfile, basically copying the whole raster

Reply:
Yes, I thought that too at some point, but looking at the code of rasterio.mask.mask, I believe you can manage that with geometry_mask: https://github.com/rasterio/rasterio/blob/1.1.3/rasterio/mask.py#L108-L109

Reply:
Or actually geometry_window: https://github.com/rasterio/rasterio/blob/830fe9d59abe0040e6ac2a8af1db4ebc7eef3676/rasterio/features.py#L379

Reply:
Oh, but geometry_window requires a dataset... It's frustrating that there is not a function that would work with just transform and shape.

Reply:
What about this? https://github.com/rasterio/rasterio/blob/830fe9d59abe0040e6ac2a8af1db4ebc7eef3676/rasterio/windows.py#L255
(Sorry for the pile of message!)

Reply:
That looks great! Thanks for finding it! I think I looked for quite a while but ended up giving up on the search.

Reply:
Wouldn't geometry_mask work?

Reply:
Hmm, I checked `geometry_mask` (which basically just calls `rasterize`). It's a bit unnecessarily complex because rasterization is not needed to just get the min/max bounds of the features. I'd prefer to not use `geometry_mask` in favour of speed.